### PR TITLE
feat: trigger API extended to fetch sheet data

### LINKED ISSUE #25645: [Feature]: Extend trigger API to fetch sheet data
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

1. In order to show sheet data in a table on datasource review page, like the designs shown below, I should be able to call an API to fetch the data

2. We have `/trigger`API which is used for fetching list of spreadsheets, list of sheets names, we can extend the same API to fetch the sheet data

[Designs](https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?node-id=1274%3A63719&mode=dev)

### Why should this be worked on?

As a frontend engineer, I should be able to fetch sheet data so that it can be displayed on datasource review page.

### PR DESCRIPTION
**Pull request by sneha122**
## Description
This PR extends the trigger API call, it introduces another trigger method called `TRIGGER_SHEET_DATA`, This method allows us to fetch sheet data which we need to show on datasource review page as shown below:
Screenshot 2023-07-25 at 2 42 35 PM
#### PR fixes following issue(s)
Fixes #25645 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- New feature (non-breaking change which adds functionality)
>
>
>
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-plugins/googleSheetsPlugin/src/test/java/com/external/config/RowsGetMethodTest.java [hunk 1]
```diff
@@ -168,4 +171,76 @@ public void transformResponse() throws JsonProcessingException {
         assertEquals(3, result.size());
         assertEquals(0, result.get(0).get(FieldName.ROW_INDEX).asInt());
     }
+
+    @Test
+    public void testvalidateExecutionMethodRequest_noSpreadsheetId_returnsException() throws JsonProcessingException {
```
Comment (sumitsum) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/test/java/com/external/config/RowsGetMethodTest.java:176]: Nit: should `v` be in caps like. `testValidateEx...` ? (for other test cases as well).

**Review by sumitsum — APPROVED** [review]

**Review by sumitsum — APPROVED** [review]

**Comment by sneha122** [comment]
@prapullac These are only server side changes for google sheets schema, and we may not able to test these without frontend changes, In that case can we go ahead and merge this PR? Post frontend changes, we can test these as well. I will add description on how to test these in FE PR

CC: @rohan-arthur

**Comment by rohan-arthur** [comment]
@sneha122 if there's no risk of regression, we should just go ahead and merge

**Comment by btsgh** [comment]
Regression tested the following areas on the DP - 

1. Google sheets - RWD all sheets - fetch many - run query - bind to table - passed
2. Google sheets - RWD all sheets - insert one - run query - passed
3. Google sheets - RWD selected sheets - fetch many - run query - bind to table - passed
4. Google sheets - RWD selected sheets - delete one - run query - row deleted - passed
5. Google sheets - R all sheets - fetch many - run query - bind to table - passed
6. Google sheets - R all sheets - insert one - run query - query fails due to scope error [as per expectation]

No issues found - works fine

**Pull request merged**